ui/appui-react/src/appui-react/statusfields/message-center/MessageCenterField.tsx

```diff
@@ -0,0 +1,162 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Notification
+ */
+import * as React from "react";
+import {
+  Button,
+  NotificationMarker,
+  Popover,
+  Tabs,
+} from "@itwin/itwinui-react";
+
+import { SvgChat } from "@itwin/itwinui-icons-react";
+import { Icon } from "@itwin/core-react";
+import { UiFramework } from "../../UiFramework";
+import { OutputMessagePriority } from "@itwin/core-frontend";
+import { MessageCenterMessage } from "./MessageCenterMessage";
+import { MessageManager } from "../../messages/MessageManager";
+import { TitleBar } from "../../layout/footer/dialog/TitleBar";
+import type { MessageType } from "@itwin/core-react";
+import type { NotifyMessageDetailsType } from "../../messages/ReactNotifyMessageDetails";
+import "./MessageCenterField.scss";
+
+/**
+ * Type for Status state to satisfy NotificationMarker type checking
+ */
+type NotificationMarkerStatus = "primary" | "negative" | "positive";
+
+/** Message Center Field React component.
+ * @public
+ */
+export function MessageCenterField() {
+  const [messages, setMessages] = React.useState(MessageManager.messages);
+  const [notify, setNotify] = React.useState(false);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [status, setStatus] =
+    React.useState<NotificationMarkerStatus>("primary");
+
+  const indicatorRef = React.createRef<HTMLButtonElement>();
+  const title = UiFramework.translate("messageCenter.messages");
+
+  const handleOpenChange = (isOpenState: boolean) => {
+    setNotify(false);
+    setIsOpen(isOpenState);
+  };
+
+  const isProblemStatus = (message: NotifyMessageDetailsType): boolean => {
+    // See priority values in DgnPlatform defined in NotificationManager
+    return (
+      message.priority === OutputMessagePriority.Error ||
+      message.priority === OutputMessagePriority.Fatal
+    );
+  };
+
+  const determineStatus = () => {
+    const message = [...MessageManager.messages].pop();
+
+    if (message) {
+      if (isProblemStatus(message)) {
+        return setStatus("negative");
+      } else if (message.priority === OutputMessagePriority.Success) {
+        return setStatus("positive");
+      } else {
+        return setStatus("primary");
+      }
+    }
+    return;
+  };
+
+  React.useEffect(() => {
+    MessageManager.registerAnimateOutToElement(indicatorRef.current);
+
+    return MessageManager.onMessagesUpdatedEvent.addListener(() => {
+      messages.length > 0 ? setNotify(true) : setNotify(false);
```

Comment:
I guess this will use old (it is updated just after this) `messages` value? Is this how it is supposed to be?

Reply:
That's a nice catch!
We got lucky, since state uses the same array reference.
```suggestion
      MessageManager.messages.length > 0 ? setNotify(true) : setNotify(false);
```